src/query/manager.ts

```diff
@@ -140,7 +140,7 @@ export class QueryManager implements Startable, Initializable {
       }
 
       // make sure we don't get trapped in a loop
-      const peersSeen = new Set()
+      const peersSeen = new Set<string>()
```

Comment:
You can also use `PeerSet` from [@libp2p/peer-collections](https://www.npmjs.com/package/@libp2p/peer-collections) for this.